util/topsql/reporter/single_target.go

```diff
@@ -119,17 +121,17 @@ func (ds *SingleTargetDataSink) run() (rerun bool) {
 }
 
 func (ds *SingleTargetDataSink) tryRegister(addr string) error {
-	if addr == "" && ds.registered {
+	if addr == "" && ds.registered.Load() {
 		ds.registerer.Deregister(ds)
-		ds.registered = false
+		ds.registered.Store(false)
 		return nil
 	}
```

Comment:
BTW, the behavior is to deregister when addr is `""`?
If so, how about just adding a `Deregister` method for `SingleTargetDataSink`?

Reply:
Ok